frontend/src/lib/lemon-ui/LemonTree/LemonTree.tsx

```diff
@@ -717,21 +775,23 @@ const LemonTree = forwardRef<LemonTreeRef, LemonTreeProps>(
             },
         })
         const sensors = useSensors(mouseSensor, touchSensor)
+        const typeAheadTimeoutRef = useRef<NodeJS.Timeout>()
 
         // Scrollable container
         const containerRef = useRef<HTMLDivElement>(null)
-
         // Current state (when matching defaultSelectedFolderOrNodeId)
         const [selectedId, setSelectedId] = useState<string | undefined>(defaultSelectedFolderOrNodeId)
         const [hasFocusedContent, setHasFocusedContent] = useState(false)
         const [isDragging, setIsDragging] = useState(false)
         const [activeDragItem, setActiveDragItem] = useState<TreeDataItem | null>(null)
-
-        // Add new state for type-ahead
+        const [disableKeyboardInput, setDisableKeyboardInput] = useState(false)
         const [typeAheadBuffer, setTypeAheadBuffer] = useState<string>('')
-        const typeAheadTimeoutRef = useRef<NodeJS.Timeout>()
-        const [isNodeTreeContextMenuOpen, setIsNodeTreeContextMenuOpen] = useState(false)
 
+        const escapeId = useCallback((id: string) => {
+            return CSS.escape(id)
+        }, [])
```

Comment:
I don't think we need a callback here. Callbacks just provide stable function references (so you can compare `currentEscapeId === previousEscapeId`), but they don't cache or memoize the results. You could just call `CSS.escape()` natively below instead of `escapeId`, and it'd do the same but faster.